Added microsoft-atp-get-machine-alerts and microsoft-atp-get-machine-users

ckaadic opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Added microsoft-atp-get-machine-alerts -> [Source](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/get-machine-related-alerts?view=o365-worldwide)
Added microsoft-atp-get-machine-users -> [Source](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/get-machine-log-on-users?view=o365-worldwide)
Ran demisto-sdk validate on the pack and fixed all the errors, resulting in updated docker image

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=18453) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=18453) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: ckaadic<br/>:x: cedric<br/><hr/>**cedric** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=18453) it.</sub>

ckaadic commented:
My local validation (demisto-sdk 1.6.3) shows only pa126 for pack_metadata, not pa114 -> I did not modify the file and as it is only a warning, I did not change it.

MLainer1 commented:
@ckaadic Hi, make sure to look at the steps that failed and fix them :)

ckaadic commented:
Fixed the failed steps. For licensing: this is both me - I linked the second username to my github account. Also tried to rebase and change the username, but it seems not to be recognized by the pipeline.

MLainer1 reviewed (changes requested):
Good work! Please add tests to test the 2 new commands, and add them to the test-playbook as well.
Make sure to update the integration README with the new commands.
In the release notes you've stated that there is a change in script as well, but it was not pushed to the pr. Is it a mistake?

ckaadic commented:
Thanks for the recommendations. Also added the readme part. I never mentioned script, but I had the new Docker version as part of the Release Notes, but it was updated in the meantime, so I removed it from the Release Notes. Will add the tests and push everything afterwards

ckaadic commented:
contribution_form_filled_verification error -> Community Flag was removed by @MLainer1
Run Validations error -> I am note the contributer of the command, fields and playbooks and did not edit those

MLainer1 commented on Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:
```diff
@@ -4711,6 +4280,12 @@ script:
       name: hostname
       required: false
       secret: false
+    - default: true
+      description: ''
+      isArray: true
+      name: endpoint
+      required: true
+      secret: false
```
I believe this is the source of the validation error, as adding a required new argument can break the backward compatibility. Can you look into it and remove if it was by mistake?

ckaadic replied:
Removed this part. Did not touch this command, but it maybe was a "fix" by demisto-sdk format as I ran this before commiting

MLainer1 commented on Packs/MicrosoftDefenderAdvancedThreatProtection/IncidentFields/incidentfield-Microsoft_365_Defender_evidencecreationtime.json:
```diff
@@ -1,31 +1,31 @@
-  "sla": 0,
-  "threshold": 72,
-  "fromVersion": "5.5.0"
+    "id": "incident_microsoftdefenderforendpointevidencecreationtime",
```
Restore the original file to avoid pushing unnecessary file updates

ckaadic replied:
Reverted this part. Did not touch the file myself, but it maybe was a "fix" by demisto-sdk format as I ran this before commiting

MLainer1 commented on Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_test.py:
```diff
@@ -2178,3 +2195,65 @@ def test_build_common_files_query(self):
             assert actual == expected
+
+
+def test_get_machine_users_command(mocker):
```
Please add a “Given, When, Then” to the test, example can be found in the other tests in the file

ckaadic replied:
Added given, when then

MLainer1 commented on Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_test.py:
```diff
@@ -2178,3 +2195,65 @@ def test_build_common_files_query(self):
+    }]
+
+
+def test_get_machine_alerts_command(mocker):
```
Please add a “Given, When, Then” to the test, example can be found in the other tests in the file

ckaadic replied:
Added given, when then

ckaadic commented on Packs/MicrosoftDefenderAdvancedThreatProtection/IncidentFields/incidentfield-Microsoft_365_Defender_evidencetype.json:
```diff
@@ -1,31 +1,31 @@
-  "sla": 0,
-  "threshold": 72,
-  "fromVersion": "5.5.0"
+    "id": "incident_microsoftdefenderforendpointevidencetype",
```
Reverted this part. Did not touch the file myself, but it maybe was a "fix" by demisto-sdk format as I ran this before commiting

ckaadic commented:
@MLainer1 it did not work. Also tried to remove the blank line. Did not work.

ckaadic commented:
Tried to use the excluding items from the documentation [here](https://xsoar.pan.dev/docs/documentation/release-notes#excluding-items), still getting the same errors in Validations

MLainer1 commented on Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/examples.txt:
```diff
@@ -68,3 +68,4 @@ microsoft-atp-advanced-hunting-cover-up query_purpose=compromised_information us
 microsoft-atp-advanced-hunting-cover-up query_purpose=connected_devices username=demisto
 microsoft-atp-advanced-hunting-cover-up query_purpose=action_types username=demisto
 microsoft-atp-advanced-hunting-cover-up query_purpose=common_files  username=demisto
+microsoft-atp-live-response-cancel-action machine_action_id=11a86b87-12b8-423b-9e8d-9775ab2da78f comment="testing"
```
I've added a command example to this file. Please change it if it's not accurate, and run the *generate-docs* command

MLainer1 commented:
Let's schedule a demo, I'll send you an email

MLainer1 commented:
@ckaadic Hi, make sure to address the conflicts

ckaadic commented:
@MLainer1 it looks like it is a Microsoft bug, as the indicator with Type CertificateThumbprint is created/updated but will always return 500 internal error.  I will remove the changes for type CertificateThumbprint as I can not fix it on my end

MLainer1 commented on Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:
```diff
@@ -1899,6 +1900,40 @@ def cancel_action(self, action_id, request_body):
+        try:
+            response = self.ms_client.http_request(method="GET", url_suffix=cmd_url)
+        except Exception:
+            raise Exception(f"Machine {machine_id} was not found")
```
Is this the only exception that can be raised from that command?

ckaadic replied:
Follow [Microsoft Documentation](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/get-machine-log-on-users?view=o365-worldwide) it is: 

> If successful and device exists - 200 OK with list of [user](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/user?view=o365-worldwide) entities in the body. If device was not found - 404 Not Found.

MLainer1 commented on Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:
```diff
@@ -1899,6 +1900,40 @@ def cancel_action(self, action_id, request_body):
+        try:
+            response = self.ms_client.http_request(method="GET", url_suffix=cmd_url)
+        except Exception:
+            raise Exception(f"Machine {machine_id} not found")
```
Is this the only exception that can be raised from that command?

ckaadic replied:
Follow [Microsoft Documentation](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/get-machine-related-alerts?view=o365-worldwide) it is:

> If successful and device exists: 200 OK with list of [alert](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/alerts?view=o365-worldwide) entities in the body. If device was not found: 404 Not Found.
